ding_availability.module

```diff
@@ -206,6 +227,39 @@ function ding_availability_holdings($provider_ids) {
 }
 
 /**
+ * Generate a human-readable amount of wait time.
+ *
+ * @param int $date
+ *   Loan date timestamp.
+ * @return string
+ *   Sane display of time to wait (months/weeks/days).
+ */
+function ding_availability_holding_format_time($date) {
+  $days = ceil(($date - time()) / 86400);
+
+  $plurals = array('day' => t('days'), 'week' => t('weeks'), 'month' => t('months'));
+  if ($days > 35) {
+    $days = ceil($days / 30);
+    $label = 'month';
+  }
+  elseif ($days > 6) {
+    $days = ceil($days / 7);
+    $label = 'week';
+  }
+  else {
+    $label = 'day';
+  }
+
+  if ($days > 1) {
+    $label = $plurals[$label];
+  }
+
+  $text = t('The waiting time is currently: <strong>@days @label</strong>', array('@days' => $days, '@label' => $label));
+
+  return $text;
```

Comment:
See above.

Also: How about using [`format_interval`](https://api.drupal.org/api/drupal/includes%21common.inc/function/format_interval/7) for this?